### PR TITLE
Detective Scanner Update

### PR DESCRIPTION
**Pull request by DaveTheHeadcrab**
The detective scanner can now search records by name, DNA, or fingerprint.
If a result is found it will display all three of those to the person.

Printing reports has been moved to a verb accessible via the objects tab or right click.

Changes the icon to something that looks a little less like a box fan: (Credit to Goon)
detectivescanner

:cl: DaveTheHeadcrab
rscadd: Adds a new icon for the detective's scanner.
tweak: Detectives scanner now has access to DNA and fingerprint records.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Now detective will never return to his office and can live out his dream of being a drunk 24/7.

File: code/modules/detective_work/scanner.dm [hunk 1]
```diff
@@ -15,9 +16,56 @@
 	var/list/log = list()
 
 /obj/item/device/detective_scanner/attack_self(var/mob/user)
+	var/search = input(user, "Enter name, fingerprint or blood DNA.", "Find record", "")
+
+	if (!search || user.stat)
```
Comment (tigercat2000) [inline code/modules/detective_work/scanner.dm]: check `user.incapacitated()`

File: code/modules/detective_work/scanner.dm [hunk 2]
```diff
@@ -15,9 +16,56 @@
...
+	if(name && fingerprint && dna)
+		to_chat(user, "<span class='notice'>Match found in station records: <b>[name]</b></span><br>\
+		<i>Fingerprint:</i><span class='notice'> [fingerprint]</span><br>\
+		<i>Blood DNA:</i><span class='notice'> [dna]</span>")
+		return
+
+	to_chat(user, "<span class='warning'>No match found in station records.</span>")
+
+
+/obj/item/device/detective_scanner/verb/print_scanner_report()
```
Comment (Fox-McCloud) [inline code/modules/detective_work/scanner.dm]: Maybe an action button instead? Verbs are icky (and yes, I know there's still plenty around). Will await @TheDZD's thought on this one.

Comment (DaveTheHeadcrab) [inline code/modules/detective_work/scanner.dm]: Seemed like a pretty inane thing to clutter the action bar up with to me.

**Comment by DaveTheHeadcrab** [comment]
@Fox-McCloud 
Already possible, they just run around with the laptops 24/7

**Comment by TheDZD** [comment]
Changelog

**Comment by DaveTheHeadcrab** [comment]
You're a changelog, DZD.

File: code/modules/detective_work/scanner.dm (hunk 2 above)
Comment (TheDZD) [inline code/modules/detective_work/scanner.dm]: Either one really works, considering this isn't something that is so time-sensitive.

**Comment by FlattestGuitar** [comment]
It's pretty neat. Normally I run with a laptop if I'm a detective and a lot of other players do that too. This doesn't make that much of a difference, maybe 5 seconds per fingerprint/DNA tops.

Quality of Life feature, :+1:

**Comment by CrAzYPiLoT-SS13** [comment]
I kinda don't like the sprite, but it's certainly better than what we have now. :+1:

**Comment by Fox-McCloud** [comment]
>Already possible, they just run around with the laptops 24/7

Shhh, don't ruin the dream, @DaveTheHeadcrab,  don't ruin the dream.

**Pull request merged**